source/blog/2016-11-28-ember-2-10-released.md

```diff
@@ -0,0 +1,223 @@
+version 2.10.0.
+
+This also kicks off the 2.11 beta cycle for all sub-projects. We encourage our
+community (especially addon authors) to help test these beta builds and report
+any bugs before they are published as a final release in six weeks' time. (The
+[ember-try](https://github.com/ember-cli/ember-try) addon is a great way to do
+that.)
+
+You can read more about our general release process here:
+
+- [Release Dashboard](http://emberjs.com/builds/)
+- [The Ember Release Cycle](http://emberjs.com/blog/2013/09/06/new-ember-release-process.html)
+- [The Ember Project](http://emberjs.com/blog/2015/06/16/ember-project-at-2-0.html)
+- [Ember LTS Releases](http://emberjs.com/blog/2016/02/25/announcing-embers-first-lts.html)
+
+---
+
+## Ember.js
+
+Ember.js is the core framework for building ambitious web applications.
+
+### Changes in Ember.js 2.10
+
+...
+
+For more details on the changes in Ember.js 2.10, please review the
+[Ember.js 2.10.0 release page](https://github.com/emberjs/ember.js/releases/tag/v2.10.0).
+
+### Upcoming changes in Ember.js 2.11
+
+...
+
+
+
+For more details on the upcoming changes in Ember.js 2.11, please review the
+[Ember.js 2.11.0-beta.1 release page](https://github.com/emberjs/ember.js/releases/tag/v2.11.0-beta.1).
+
+---
+
+## Ember Data
+
+Ember Data is the offical data persistence library for Ember.js applications.
+
+### Changes in Ember Data 2.10
+
+Ember Data 2.10 represents the work of over 11 direct contributors and over
+27 commits.
+
+There have been no user facing changes in Ember Data 2.10. Under the
+hood, [@runspired](https://github.com/runspired) has instrumented
+Ember Data with some internal tools to make it easier to track the
+performance of Ember Data. This instrumentation is stripped from the
+production build of Ember Data and should have no impact on existing
+applications that rely on Ember Data.
+
+[@runspired](https://github.com/runspired) also
+[improved the messaging](https://github.com/emberjs/data/pull/4578)
+around assertions and warnings when an adapter responds with no `id`
+for a record.
+
+For more details on the changes in Ember Data 2.10, please review the
+[Ember Data 2.10.0 release page](https://github.com/emberjs/data/releases/tag/v2.10.0).
+
+### Upcoming changes in Ember Data 2.11
+
+Ember Data 2.11 is already shaping up to be a big release with many
+performance and documentation improvements.
+
+#### HasMany Computed Property Keys
+
+One of the performance improvements included in the Ember Data
+2.11.0-beta.1 release fixes an issue where Ember Data was doing too
+much work when records in a `DS.hasMany()` changed. As a result of
+this extra work, Ember Data would incorrectly invalidate computed
+properties that depended on these `hasMany` relationships. After this
+fix you may need to update computed property keys that depended on
+`record.hasManyRelationshipName` to `record.hasManyRelationshipName.[]`.
+
+#### Deprecate store.serialize
```

Comment:
My comments from before seemed to get lost so I'll redo them quick. 2.11 also deprecates:

- `store.lookupAdapter` and `store.lookupSerializer` as they are simply proxies for `store.adapterFor` and `store.serializerFor`. These methods were never fully public or private.
(changes: https://github.com/emberjs/data/blob/master/addon/-private/system/store.js#L2595-L2609)

- `store.recordIsLoaded` is deprecated in favor of `store.hasRecordForId`, as these methods have been identical for sometime. https://github.com/emberjs/data/pull/4665